snap/epoch.go

```diff
@@ -212,6 +204,44 @@ func (e *Epoch) String() string {
 	return string(buf)
 }
 
+// CanRead checks whether this epoch can read the data written by the
+// other one.
+func (e *Epoch) CanRead(other *Epoch) bool {
+	// the intersection between e.Read and other.Write needs to be non-empty
+
+	// normalize (empty epoch should be treated like "0" here)
+	var rs, ws []uint32
+	if e != nil {
+		rs = e.Read
+	}
+	if other != nil {
+		ws = other.Write
+	}
+	if len(rs) == 0 {
+		rs = []uint32{0}
+	}
+	if len(ws) == 0 {
+		ws = []uint32{0}
+	}
+
```

Comment:
Is there something that ensures that `rs` and `ws` are indeed sorted?

Reply:
Either by construction (when the epoch is a string), or from `snap.(*Epoch).Validate()`, when the epoch is more complex.
In tests, you can build an invalid epoch, but in the wild they all come from one of those two :-)